packages/mui-material/src/internal/SwitchBase.js

```diff
@@ -166,41 +169,77 @@ const SwitchBase = React.forwardRef(function SwitchBase(props, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      input: inputProps,
+      ...slotProps,
+    },
+  };
+
+  const [RootSlot, rootSlotProps] = useSlot('root', {
+    ref,
+    elementType: SwitchBaseRoot,
+    className: clsx(classes.root, className),
```

Comment:
Nit: We could remove `className` from here and from the props destructuring (as it's not used elsewhere) and allow it to be forwarded through `other`. This will also allow us to remove the `clsx` import.

Reply:
good one.